ThunderCloud/AuthenticationController.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  AuthenticationController.swift
+//  ThunderCloud
+//
+//  Created by Simon Mitchell on 15/02/2018.
+//  Copyright © 2018 threesidedcube. All rights reserved.
+//
+
+import Foundation
+import ThunderRequest
+
+/// The Authentication Controller is responsible for authenticating with the
+/// Storm CMS API. Any endpoint that requires an authorisation token can use
+/// this controller to obtain a token.
+public class AuthenticationController {
+	
+	/// A structual representation of the user's authentication
+	public struct Authorization: Decodable, Encodable {
+		
+		/// The authorization token
+		let token: String
+		
+		/// The expiry date of the Authorization
+		let expiry: Date
+		
+		/// Whether the user is currently authenticated
+		var hasExpired: Bool {
+			return expiry < Date()
+		}
+	}
+	
+	/// A typealias for an authentication completion
+	public typealias AuthenticationCompletion = (_ authorization: Authorization?, _ error: Error?) -> Void
+	
+	private let requestController: TSCRequestController
+	
+	public init() {
+		
+		requestController = TSCRequestController(baseAddress: "https://auth.cubeapis.com/v1.5")
```

Comment:
Should be `v1.6`

Reply:
I'm not sure if this is the case, I believe it's a different API with different versioning to the CMS content API

Reply:
`https://auth.cubeapis.com/latest` says v1.6 is the latest. I remember Steve asking me why we were still using v1.5 a while back. Have you tried it and seen if it works?

Reply:
Will give it a go now! My bad, should've checked 👍🏻